### PR TITLE
Fix `Control::set_global_position` for rotated/scaled transforms

### LINKED ISSUE #85069: Control node has unexpected movement behavior when rotated
### Godot version

v4.1.3.stable.official

### System information

Fedora Linux 38 (Workstation Edition) - v4.1.3.stable.official - Compatibility (GLES3)

### Issue description

The control node has inconsistent movement behavior when it's rotated while having it's global position changed.
For example it might move diagonally, or it has higher speed than expected.

This only happens when it's global position variable gets modified.

It works as expected when rotation_degrees is set to zero or when the position variable gets updated.

The control node is supposed to move similar to how a sprite 2D moves when it's position gets changed.

This also happens with Godot 4.2 RC 1

### Steps to reproduce

Make a root scene (This can be any node) and add a Control node. For visualization add a Colored rect as the child of the Control node.

Attach a script, and have it move the Control node's global position downwards by one, every frame.
Code example below this line:
```extends Node

func _physics_process(delta: float) -> void:
	$Control.global_position.y += 1 * delta
```

Go back to the editor and set the Control node's rotation to 1 in the inspector.

Then run the scene.

Reproduction

### Minimal reproduction project
Open master.tscn and run

[ControlNodeBugReproduction.zip](https://github.com/godotengine/godot/files/13404261/ControlNodeBugReproduction.zip)

### LINKED ISSUE #89497: Setting Rotation and GlobalPosition for Control causes runaway Position issues.
### Tested versions

- Reproducible in: v4.3.dev4.mono.official [df78c0636]

### System information

Godot v4.3.dev4.mono - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1080 (NVIDIA; 31.0.15.5161) - 13th Gen Intel(R) Core(TM) i5-13600KF (20 Threads)

### Issue description

Setting the GlobalPosition of a control while it's Rotation is non-zero can cause runaway issues with the controls position.

https://github.com/godotengine/godot/assets/1730988/437decae-d74f-44f8-83b3-3e18d1f12c01

### Steps to reproduce

In the repro, hold down left mouse button to see the issue happening.

### Minimal reproduction project (MRP)

[ControlPositionRotationBug.zip](https://github.com/godotengine/godot/files/14609095/ControlPositionRotationBug.zip)

### PR DESCRIPTION
**Pull request by kleonc**
Not sure how I've tested #87432 that I haven't noticed this but the calculations in there were correct only for `get_internal_transform().get_origin()` being zero. Redone the calculations (and kinda left them in comments), now they should be correct. (cc @Rindbee)

Fixes #89497.
Fixes #85069.

### CODE REVIEW

**Review by Rindbee — APPROVED** [review]
Yes, now it is correct.

File: scene/gui/control.cpp [hunk 1]
```diff
@@ -1406,15 +1406,11 @@ void Control::_set_global_position(const Point2 &p_point) {
...
-	}
-
-	Point2 internal_position = global_transform_cache.affine_inverse().xform(p_point);
-
-	set_position(internal_position + data.pos_cache, p_keep_offsets);
+	// (parent_global_transform * T(new_position) * internal_transform).origin == new_global_position
+	// (T(new_position) * internal_transform).origin == new_position_in_parent_space
+	// new_position == new_position_in_parent_space - internal_transform.origin
+	Point2 position_in_parent_space = data.parent_canvas_item ? data.parent_canvas_item->get_global_transform().affine_inverse().xform(p_point) : p_point;
+	set_position(position_in_parent_space - _get_internal_transform().get_origin(), p_keep_offsets);
```
Comment (Sauermann) [inline scene/gui/control.cpp:1413]: I would have to check in detail, but isn't this missing the handling of [CanvasItem.top_level](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-property-top-level)?

Comment (Rindbee) [reply scene/gui/control.cpp:1413]: `data.parent_canvas_item` is `nullptr` in that case.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**